test/unit/signup/account_manager_test.rb

```diff
@@ -126,23 +126,29 @@ class ProviderAccountManagerTest < Signup::AccountManagerTest
       assert_equal 'API', account.first_service!.name
     end
 
-    test 'creating a provider will create default Backend if api_as_product is disabled' do
+    test 'creating a provider will create default Backend regardless of api_as_product' do
       Account.any_instance.stubs(:provider_can_use?).returns(false)
       Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(false)
       account = signup_account_manager.create(signup_params).account
-      assert_equal BackendApi.default_api_backend, account.default_service.api_backend
+      assert_expected_backend_api(account)
     end
 
     test 'creating a provider will create default Backend if api_as_product is enabled' do
       Account.any_instance.stubs(:provider_can_use?).returns(false)
       Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(true)
       account = signup_account_manager.create(signup_params).account
-      assert account.backend_apis.first
-      assert_equal BackendApi.default_api_backend, account.default_service.api_backend
+      assert_expected_backend_api(account)
     end
 
     private
 
+    def assert_expected_backend_api(account)
+      assert_equal 1, account.backend_apis.count
+      assert(service = account.default_service)
+      assert_equal 1, service.backend_apis.count
+      assert_equal BackendApi.default_api_backend, service.api_backend
+    end
+
```

Comment:
This is not needed. The end result of APIAP and non-APIAP scenarios may be the same, but how we validate that in the test should be different.

Reply:
Why it should be different? we want the exact same result, no? so why should we write the same twice? :confused:

Reply:
It's about abstracting implementation details. For APIAP we need to validate that a backend api was created. For non-APIAP what's important is if the service has a given private endpoint set. We don't care about which type of object is caring that info in the latter.

If you will, keep the two test cases as they were and describe them
- "First service has a private endpoint" (This means it is functional for non-APIAP accounts)
- "First backend API is created" (This means https://github.com/3scale/porta/pull/1292 fixed the bug it was meant to)

Reply:
Ok, I understand you now :+1: